File: Algorithm/10주_완성_알고리즘_코딩테스트_기초/2.알고리즘의_연산량/2C_다양성/2c.py
# 2C 다양성

n = int(input())
inputData = list(map(int, input().split(' ')))

result = 1
current = inputData[0]

for i in range(1, n):
    if current != inputData[i]:
        result += 1
        current = inputData[i]

print(result)

File: Algorithm/10주_완성_알고리즘_코딩테스트_기초/2.알고리즘의_연산량/2C_다양성/2c.js
// 2C 다양성

const fs = require('fs');
const inputData = fs.readFileSync('/dev/stdin').toString().split('\n');

const n = Number(inputData[0]);
const array = inputData[1].split(' ').map(Number);

let result = 1;
let current = array[0];

for (let i=1; i<array.length; i++) {
  if (current !== array[i]){
    result += 1;
    current = array[i];
  }
}

console.log(result);
